component/portal/src/main/java/org/exoplatform/portal/mop/rest/NavigationRest.java

```diff
@@ -40,46 +64,58 @@ public NavigationRest(UserPortalConfigService portalConfigService) {
     this.portalConfigService = portalConfigService;
   }
 
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed("users")
+  @ApiOperation(value = "Gets navigations", httpMethod = "GET", response = Response.class, notes = "This returns site navigations")
+  @ApiResponses(value = { @ApiResponse(code = 200, message = "Request fulfilled"),
+      @ApiResponse(code = 400, message = "Invalid query input"), @ApiResponse(code = 404, message = "Navigation does not exist"),
+      @ApiResponse(code = 500, message = "Internal server error") })
+  public Response getSiteNavigation(@Context HttpServletRequest request) {
+    HttpUserPortalContext userPortalContext = new HttpUserPortalContext(request);
+    String portalName = portalConfigService.getDefaultPortal();
+
+    ConversationState state = ConversationState.getCurrent();
+    Identity userIdentity = state == null ? null : state.getIdentity();
+    String username = userIdentity == null ? null : userIdentity.getUserId();
+    if (StringUtils.equals(username, IdentityConstants.ANONIM)) {
+      username = null;
+    }
+
+    try {
+      UserPortalConfig userPortalConfig = portalConfigService.getUserPortalConfig(portalName, username, userPortalContext);
+      List<ResultUserNavigation> allNavs = userPortalConfig.getUserPortal()
+                                                           .getNavigations()
+                                                           .stream()
+                                                           .filter(userNavigation -> !userNavigation.getKey()
+                                                                                                    .getName()
+                                                                                                    .equals("global"))
+                                                           .map(userNavigation -> new ResultUserNavigation(userNavigation))
```

Comment:
This can really be a very big object, especially for someone who belongs to a lot of spaces, isn't it ?
We may need a pagination here.

Reply:
Yes it is possible.
This improvment will be tracked and done in another task